spec/requests/admin/form_templates_controller_spec.rb

```diff
@@ -163,4 +163,75 @@
       end
     end
   end
+
+  describe "#preview" do
+    fab!(:tag1, :tag)
+    fab!(:tag2, :tag)
+    fab!(:tag3, :tag)
+    fab!(:tag_group) { Fabricate(:tag_group, tags: [tag1, tag2, tag3]) }
+    let!(:form_template_valid) { <<~YAML }
+        - type: tag-chooser
+          id: tag-chooser
+          tag_group: "#{tag_group.name}"
+          attributes:
+            none_label: "Select an item"
+            label: "Enter label here"
+            multiple: true
+      YAML
+    let!(:form_template_invalid) { <<~YAML }
+        - type: input
+          id: dumplicated
+          attributes:
+            label: "label"
+            placeholder: "placeholder"
+        - type: input
+          id: dumplicated
+          attributes:
+            label: "label"
+            placeholder: "placeholder"      
+      YAML
+
+    context "when logged in as an admin" do
+      before { sign_in(admin) }
+
+      it "delegates validate and process to the template" do
+        template = mock
+        FormTemplate
+          .expects(:new)
+          .with(name: "test", template: form_template_valid)
+          .returns(template)
+        template.expects(:validate!)
+        template.expects(:process!)
+
+        get "/admin/customize/form-templates/preview.json",
+            params: {
+              name: "test",
+              template: form_template_valid,
+            }
+      end
```

Comment:
```suggestion
```

This test case is mostly testing internally implementation details of the controller's action which is not really the concerns for requests spec. We are mostly concerned about the response status and the body in these tests.